### PR TITLE
Explicitly manage the working directory to support multiple input files

### LINKED ISSUE #82: Fails on processing multiple files
When passing multiple input files to either `validate` or `render` , the templexer fails on the second file due to the change of working directory for the first file.

### PR DESCRIPTION
**Pull request by hendrikmaus**
Fixes #82.

This changeset refactors the way we manage the current working directory of the templexer when it processes a file.
We start by storing the initial working dir, then we switch to each file and back afterwards.

### CODE REVIEW

**Comment by Kenec** [comment]
Tested with different config paths and it works. Should we also add a unit test for the `switch_working_directory()` function?

**Review by Kenec — APPROVED** [review]

**Comment by hendrikmaus** [comment]
> Tested with different config paths and it works. Should we also add a unit test for the `switch_working_directory()` function?

Good idea. However, it should be added as integration test to test features, not code. I'll add the test case tomorrow. Thanks for the review already.

File: tests/integration/render.rs [hunk 1]
```diff
@@ -2,16 +2,21 @@ use assert_cmd::prelude::*;
...
 use std::path::PathBuf;
 use std::process::Command;
 
 const BIN_NAME: &'static str = env!("CARGO_PKG_NAME");
 
-/// Generate a dummy configuration and dump it to a temporary location on disk
-/// this should be called at the beginning of each test case and concluded with calling `drop_temp_folder`.
-fn generate_config() -> Result<PathBuf, io::Error> {
-    let config = r#"---
+struct Config {
```
Comment (hendrikmaus) [inline tests/integration/render.rs:11]: @Kenec I wrapped the unique config into a struct and implemented the `Drop` trait on it. That way, Rust will call it automatically when the config gets out of scope and therefore a tear-down is implemented. No need to call any drop function.

File: tests/integration/render.rs [hunk 2]
```diff
@@ -172,54 +176,60 @@ fn pipe_output_to_a_tool_that_exists() -> anyhow::Result<()> {
...
 
     // the binary execution should fail because xyz tool doesn't exist.
     cmd.assert().failure();
 
-    drop_temp_folder(&format!("{}", config.parent().unwrap().to_string_lossy()))?;
+    Ok(())
+}
+
+#[test]
+fn render_multiple_files() -> anyhow::Result<()> {
```
Comment (hendrikmaus) [inline tests/integration/render.rs:227]: @Kenec and here goes the multi-file test case to assert the feature works as expected.

**Pull request merged**